expat/lib/xmlparse.c

```diff
@@ -1230,8 +1230,11 @@ generate_hash_secret_salt(XML_Parser parser) {
 #  endif /* ! defined(_WIN32) && defined(XML_DEV_URANDOM) */
   /* .. and self-made low quality for backup: */
 
+  entropy = gather_time_entropy();
+#ifndef __wasi__
```

Comment:
PS: This would arguably be a bit easier to read:

```suggestion
#if ! defined(__wasi__)
```